layers/+distributions/spacemacs-base/funcs.el

```diff
@@ -184,6 +184,49 @@ Dedicated (locked) windows are left untouched."
   (interactive "p")
   (spacemacs/rotate-windows (* -1 count)))
 
+(defun spacemacs/move-buffer-to-window (windownum follow-focus-p)
+  "Moves a buffer to a window, using the spacemacs numbering. follow-focus-p
+   controls whether focus moves to new window (with buffer), or stays on
+   current"
+  (interactive)
+  (let ((b (current-buffer))
+        (w (selected-window)))
+    (save-selected-window
```

Comment:
Saving the selected window, selecting another window and changing its buffers with `switch-to-buffer` is not the best way to change the buffer of a window. Partly because `switch-to-buffer` is not guaranteed to display the buffer in the selected window (you need to pass `force-same-window`). It is much better to use `set-window-buffer`:
```elisp
(let (;; ...
      (other-win (get-window-by-number windownum)))
  ;; ...
  (set-window-buffer other-win current-buf)
  ;; ...
  ))
```
(I'm assuming `get-window-by-number` or an equivalent function exists)